gen/code.ml

```diff
@@ -109,7 +111,7 @@ let seq_sd sd1 sd2 =
 let fold_ie f r = f Ext (f Int r)
 let fold_sd f r = f Diff (f Same r)
 let do_fold_extr withj f r =
-  let r = f (Dir W) (f (Dir R) (f Irr r)) in
+  let r = f (Dir W) (f (Dir R) (f (Dir D) (f (Dir I) (f Irr r)))) in
   if withj then f (Dir J) r
   else r
```

Comment:
```suggestion
  let all_dirs = [Dir W; Dir R; Dir J; Dir D; Dir I] in
  let dirs = List.filter (function Dir J -> withj | -> true) all_dirs in
  List.fold_right f dirs (Irr r)
```
Just as a minor nitpick-y suggestion: as the number of the direction values grows, it might make sense fold a list here. (Please ignore if you dislike the readability of this or double-check it compiles if you like it.)

Reply:
that definitely looks a lot better, I will try implement this in a subsequent commit

Reply:
I have made the change; it seems to work for my tests but let me know if you see anything wrong. Is it ok to resolve this comment?

Reply:
Thank you, @hugookeeffe -- of course, totally okay to resolve.